bin/commands/runs.js

```diff
@@ -144,6 +147,23 @@ module.exports = function run(args, rawArgs) {
       if (isAccessibilitySession && isBrowserstackInfra) {
         await createAccessibilityTestRun(bsConfig);
       }
+
+      if (turboScaleSession) {
+        const gridDetails = await getTurboScaleGridDetails(bsConfig);
+        Constants.turboScaleObj.gridDetails = gridDetails;
+
+        if (gridDetails.isTrialGrid) {
+          bsConfig.connection_settings.local = true;
+          bsConfig.connection_settings.local_inferred = true;
+          bsConfig.connection_settings.local_mode = 'on-demand';
+        }
+
+        Constants.turboScaleObj.gridUrl = gridDetails.cypressUrl;
+        Constants.turboScaleObj.uploadUrl = gridDetails.cypressUrl + '/upload';
+        Constants.turboScaleObj.buildUrl = gridDetails.cypressUrl + '/build';
```

Comment:
Won't this throw an `undefined` exception if the request for grid details fail in `getTurboScaleGridDetails` at https://github.com/browserstack/browserstack-cypress-cli/pull/736/files#diff-6ad671bcacc06082182a70064d2229f5537447360f11c88e1e2d92d800e9befeR59-R77

Reply:
added that handling.. re-review